main-script.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include "matrix-matrix-product.hpp"
+#include "matrix-vector-product.hpp"
+#include "read-matrix-market.hpp"
+#include "write-matrix-market.hpp"
+
+using namespace std;
+typedef vector<vector<double>> Matrix;
+typedef vector<double> Vector;
+
+int main(int argc, char* argv[]) {
+    const char* file1 = argv[1];
+    const char* file2 = argv[2];
+    int operation_type = atoi(argv[3]);
+
+    if (operation_type == 0) {
+        // Read the matrix and vector
+        Matrix m1 = read_matrix_market(file1); // Read in the matrix file.
```

Comment:
Great function name! In this case, you really don't need any comments as they can go out of sync with your code easily. But, since your function name is so good it explains exactly what's happening.